### PR TITLE
Added Image model with polymorphic relation to product

### PR DESCRIPTION
**Pull request by JuanSaieh**
Changes:
- Added Image model with a polymorphic relation to Product.
  - Product has many Images, Image has one Product
  - Image has an attachment to store image
- Added Cocoon Gem to add Image dynamically in Product form
- Updated Product show view to display Images
Screen Shot 2019-08-16 at 2 37 27 PM
Screen Shot 2019-08-16 at 2 37 10 PM
Screen Shot 2019-08-16 at 2 36 27 PM

Trello Link: https://trello.com/c/y8Mooyw3

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -47,10 +76,12 @@
     t.string "reset_password_token"
     t.datetime "reset_password_sent_at"
     t.datetime "remember_created_at"
+    t.string "roles"
```
Comment (albertomejia92) [inline db/schema.rb]: 👀

File: app/views/products/show.html.erb [hunk 2]
```diff
@@ -7,3 +7,36 @@
...
+<div class="card text-center " style="width: 23rem;">
+  <div id="carouselExampleIndicators" class="carousel slide" data-ride="carousel">
+    <ol class="carousel-indicators">
+      <li data-target="#carouselExampleIndicators" data-slide-to="0" class="active"></li>
+      <li data-target="#carouselExampleIndicators" data-slide-to="1"></li>
+      <li data-target="#carouselExampleIndicators" data-slide-to="2"></li>
+    </ol>
+    <div class="carousel-inner">
+      <% unless @product.images.empty? %>
+        <% (0...@product.images.count).each do |num| %>
```
Comment (albertomejia92) [inline app/views/products/show.html.erb]: When you can, please refactor this.

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -17,4 +17,5 @@
 
 //= require jquery3
 //= require popper
-//= require bootstrap-sprockets
\ No newline at end of file
+//= require bootstrap-sprockets
+//= require cocoon
```
Comment (albertomejia92) [inline app/assets/javascripts/application.js]: Add new line please.

Comment (JuanSaieh) [reply app/assets/javascripts/application.js]: Done!

File: app/views/products/show.html.erb (hunk 2 above)
Comment (JuanSaieh) [reply app/views/products/show.html.erb]: Done!

File: db/schema.rb (hunk 1 above)
Comment (JuanSaieh) [reply db/schema.rb]: Removed!

**Pull request merged**